### PR TITLE
refactor(voice): refs?

### PR DESCRIPTION
**Pull request by Dovchik**
For models any non-required/optional field is nullable

Add EmptyResponse class for empty responses from the API, refactor api calls to use it

### CODE REVIEW

File: src/Sinch/Core/Http.cs [hunk 1]
```diff
@@ -154,13 +161,27 @@ internal class Http : IHttp
...
+                           ?? throw new InvalidOperationException(
+                               $"{typeof(TResponse).Name} is null");
+
+                // if empty response is expected, any non related response is dropped
+                if (typeof(TResponse) == typeof(EmptyResponse))
+                {
+                    // if not empty content, check what is there for debug purposes.
+                    if (result.Content.GetType().Name != "EmptyContent")
+                    {
+                        _logger?.LogDebug("Response is not json, but {content}",
```
Comment (asein-sinch) [inline src/Sinch/Core/Http.cs:173]: I'm not sure about the log here: shouldn't it be something like "Response is not empty" ?

File: src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs [hunk 2]
```diff
@@ -62,6 +65,6 @@ public class CustomCalloutRequest
         ///     This could result in further SVAML or some other application logic function.
         ///     <example>"https://your-application-server-host/application"</example>
         /// </summary>
-        public string Pie { get; set; }
+        public string? Pie { get; set; }
```
Comment (asein-sinch) [inline src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs:68]: Similarly to `ace` and `ice`, the `pie` property can also be a string representation of a JsonObject

File: src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public class TtsCalloutRequest
         ///     Example:
         ///     <c>#tts[Hello from Sinch];#ssml[&lt;speak&gt;&lt;break time=\"250ms\"/&gt;Have a great day!&lt;/speak&gt;]</c>
         /// </summary>
-        public string Promts { get; set; }
+        public string? Promts { get; set; }
```
Comment (asein-sinch) [inline src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs:79]: Typo

File: src/Sinch/Voice/Calls/Call.cs [hunk 4]
```diff
@@ -60,19 +61,19 @@ public sealed class Call
         /// <summary>
         ///     An object that can be used to pass custom information related to the call.
         /// </summary>
-        public object Custom { get; set; }
+        public JsonObject? Custom { get; set; }
```
Comment (asein-sinch) [inline src/Sinch/Voice/Calls/Call.cs:64]: Why this change? The documentation mentions a `string`: https://developers.sinch.com/docs/voice/api-reference/voice/tag/Calls/#tag/Calls/operation/Calling_GetCallResult!c=200&path=custom&t=response

**Review by asein-sinch — COMMENTED** [review]
Review done. I checked the parts related to Conversation and SMS too, even if the PR mentioned voice only

File: src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs (hunk 2 above)
Comment (Dovchik) [reply src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs:68]: I'll update it in separate PR

File: src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs (hunk 3 above)
Comment (Dovchik) [reply src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs:79]: That will be in separate as well

File: src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs (hunk 2 above)
Comment (asein-sinch) [reply src/Sinch/Voice/Callouts/Callout/CustomCalloutRequest.cs:68]: I don't understand what qualifies a change to be in a PR or not. If you don't plan to update this property, are you going to revert what you did for `Ace` and `Ice` just above?

File: src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs (hunk 3 above)
Comment (asein-sinch) [reply src/Sinch/Voice/Callouts/Callout/TextToSpeechCalloutRequest.cs:79]: In this case, please create a ticket and communicate the reference to the reviewers

File: src/Sinch/Voice/Calls/Call.cs (hunk 4 above)
Comment (asein-sinch) [reply src/Sinch/Voice/Calls/Call.cs:64]: In this case, maybe you can leave it as it was and fix the type only once in a separate PR?

**Review by asein-sinch — APPROVED** [review]
